### PR TITLE
Ios/add unregister support

### PR DESCRIPTION
**Pull request by clairewli**
Add support on the iOS for unregisterEventListener and unregisterRequestHandler

### CODE REVIEW

File: ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.h [hunk 1]
```diff
@@ -39,15 +39,19 @@ NS_ASSUME_NONNULL_BEGIN
 + (void)sendRequest:(ElectrodeBridgeRequest *)request
     completionHandler:(ElectrodeBridgeResponseCompletionHandler)completion;
 
-+ (void)registerRequestHanlderWithName:(NSString *)name
++ (NSUUID *)registerRequestHanlderWithName:(NSString *)name
```
Comment (jomnius) [inline ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.h]: Possible typo, proposing to change "Hanlder" --> "Handler" (everywhere)

File: ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.h [hunk 2]
```diff
@@ -39,15 +39,19 @@ NS_ASSUME_NONNULL_BEGIN
...
               requestCompletionHandler:
                   (ElectrodeBridgeRequestCompletionHandler)completion;
 
-+ (void)addEventListnerWithName:(NSString *)name
++ (void)unregisterRequestHandlerWithUUID: (NSUUID *)uuid;
+
+
++ (NSUUID *)addEventListnerWithName:(NSString *)name
```
Comment (jomnius) [inline ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.h]: Possible typo, proposing to change "Listner" --> "Listener" (everywhere)

File: ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.m [hunk 3]
```diff
@@ -21,6 +21,43 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+@interface ElectrodeQueuedEventListener: NSObject
+@property (nonatomic, strong) NSUUID *uuid;
+@property (copy) ElectrodeBridgeEventListener listener;
```
Comment (jomnius) [inline ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.m]: Proposal: having a property with both atomic and copy seems a bit unusual, so I would propose writing the "atomic" here, too, to document that this was done on purpose.

File: ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.m [hunk 4]
```diff
@@ -21,6 +21,43 @@
...
+        _listener = listener;
+    }
+    
+    return self;
+}
+@end
+
+@interface ElectrodeQueuedRequestHandler: NSObject
+@property (nonatomic, strong) NSUUID *uuid;
+@property (copy) ElectrodeBridgeRequestCompletionHandler handler;
```
Comment (jomnius) [inline ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.m]: Proposal: having a property with both atomic and copy seems a bit unusual, so I would propose writing the "atomic" here, too, to document that this was done on purpose.

File: ios/ElectrodeReactNativeBridge/ElectrodeBridgeProtocols.h [hunk 5]
```diff
@@ -79,12 +79,19 @@ typedef void (^ElectrodeBridgeEventListener)(id _Nullable eventPayload);
...
  * Register the request handler
  * @param name name of the request
+ * @param uuid uuid of the request handler
  * @param completion call back to be issued for a given request.
  */
-- (NSUUID *)registerRequestCompletionHandlerWithName:(NSString *)name
-                                   completionHandler:
-                                       (ElectrodeBridgeRequestCompletionHandler)
-                                           completion;
+- (void) registerRequestCompletionHandlerWithName:(NSString *)name
```
Comment (jomnius) [inline ios/ElectrodeReactNativeBridge/ElectrodeBridgeProtocols.h]: OCD comment: seems like you have no coding convention for whitespace? Within 9 rows I can see "(NSString *)name", "<space>(NSString *)<space>uuid" and "<space>(NSUUID *)uuid" :) Can't remember any more how we automated this kind of things, but should be possible.

Comment (jomnius) [reply ios/ElectrodeReactNativeBridge/ElectrodeBridgeProtocols.h]: Wondering, if e.g. name should be _Nonnull

**Comment by jomnius** [comment]
General: feel free to ignore all my comments, I only wanted to raise attention to this already SIX DAYS old PR. Just want the real fix, unregistering for iOS.

File: ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.m (hunk 3 above)
Comment (clairewli) [reply ios/ElectrodeReactNativeBridge/ElectrodeBridgeHolder.m]: @jomnius Thanks for the comments. Please check apple docs here on block property example: https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/WorkingwithBlocks/WorkingwithBlocks.html

**Comment by deepueg** [comment]
looks good.!

**Pull request merged**